src/extension/appLauncher.ts

```diff
@@ -279,6 +285,32 @@ export class AppLauncher {
         });
     }
 
+    private resolveAndSaveEmulator(mobilePlatform: GeneralMobilePlatform, launchArgs: any, mobilePlatformOptions: any): Promise<void> {
```

Comment:
Let's add a Javadoc here describing that we launch an emulator only in case if needed. Also I think that this function will better look as a part of the mobile platform interface. So we'll be able to call one function with different implementations for both Android and iOS. 
With respect to the naming, I think neutral `startVirtualDevice` will look better for both platforms.